components/brave_rewards/resources/tip/lib/host.ts

```diff
@@ -233,12 +246,20 @@ export function createHost (): Host {
         dialogArgs.publisherKey,
         amount,
         kind === 'monthly' ? true : false
-      ])
-
-      stateManager.update({
-        tipProcessed: true,
-        tipAmount: amount
-      })
+      ]);
+
+      // Gemini currently has up to a 10-minute delay on the balance
+      // actually being moved from the user's account.
+      // If the tipping banner doesn't hear back within 3 seconds,
+      // we consider the transaction as non-errored and show the success UI.
+      ((amount: number) => {
```

Comment:
Comment is perfect - thanks!

The arrow function arg to `setTimeout` below already closes over the `amount` variable, so we probably don't need this immediately-invoked wrapper function, and we should be able to call `setTimeout` directly.

Reply:
Yeah, I just wanted to make sure that no changes to `amount` after calling `setTimeout()` are reflected in the call to `update()`, e.g.:
```
setTimeout(() => {
  stateManager.update({
    tipProcessed: true,
    tipAmount: amount
  })
}, 3000)

amount *= 2
```
The IIFE version doesn't suffer from this. WDYT?